csharp/src/Ice/Reference.cs

```diff
@@ -601,19 +610,20 @@ static void StartQueryOption(StringBuilder sb, ref bool firstOption)
                 OutgoingRequestFrame request,
                 bool oneway,
                 int i,
+                IReadOnlyList<InvocationInterceptor> interceptors,
                 IProgress<bool>? progress,
                 CancellationToken cancel)
             {
                 cancel.ThrowIfCancellationRequested();
-                if (i < proxy.Communicator.InvocationInterceptors.Count)
+                if (i < interceptors.Count)
```

Comment:
As commented elsewhere, I would perform the logical concatenation of per-proxy and per-communicator interceptors here.

Reply:
I was trying to avoid allocating a new list of interceptors for each request.

Reply:
I believe you could do it without creating a new list: you're just iterating over 2 lists instead of one with (for example) a single index over both.

Reply:
fixed